django/db/backends/sqlite3/creation.py

```diff
@@ -106,3 +124,35 @@ def test_db_signature(self):
         else:
             sig.append(test_database_name)
         return tuple(sig)
+
+    def setup_worker_connection(self, _worker_id):
+        settings_dict = self.get_test_db_clone_settings(_worker_id)
+        # connection.settings_dict must be updated in place for changes to be
+        # reflected in django.db.connections. Otherwise new threads would
+        # connect to the default database instead of the appropriate clone.
+
+        start_method = multiprocessing.get_start_method()
+        if start_method == "fork":
+            # update settings_dict in place
+            self.connection.settings_dict.update(settings_dict)
+            self.connection.close()
+        elif start_method == "spawn":
+            alias = self.connection.alias
+            connection_str = (
+                f"file:memorydb_{alias}_{_worker_id}?mode=memory&cache=shared"
+            )
+            source_db = sqlite3.connect(f"file:{alias}_{_worker_id}.sqlite3", uri=True)
```

Comment:
Presumably if(!) my previous comment about hard-coding `.sqlite3` applied, ideally it'd be scrubbed from here, too. But I can't see that we're otherwise examining the filename in this method, and I don't know that it _matters_...